keps/sig-node/3545-improved-multi-numa-alignment/README.md

```diff
@@ -0,0 +1,451 @@
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/cm/topologymanager`: `09-23-2022` - `92.4`
+
+##### Integration tests
+
+ These cases will be added in the existing integration tests:
+  - Feature gate enable/disable tests
+  - `prefer-closest-numa-nodes` policy option works as expected. When policy option is enabled
+     - `Merge` prefers hints with the lowest average distance between NUMA nodes.
+  - Verify no significant performance degradation
+
+##### e2e tests
+
+These cases will be added in the existing e2e tests:
+  - Feature gate enable/disable tests
+  - `prefer-closest-numa-nodes` policy option works as expected.
+
+### Graduation Criteria
+
+#### Alpha
+- [X] Implement the new policy option.
+- [X] Temporarily include NUMA distances discovery logic in kubelet code.
+- [X] Add proper e2e node tests.
+
+#### Alpha to Beta Graduation
+- [X] Gather feedback from the consumer of the `prefer-closest-numa-nodes` policy option.
+- [X] Remove NUMA distances discovery logic from kubelet in favor of updated cAdvisor.
+- [X] No major bugs reported in the previous cycle.
+
+#### Beta to G.A Graduation
+- [X] Allowing time for feedback (1 year).
+- [X] Risks have been addressed.
+- [X] Presence of E2E test validating the described behavior on 4+ NUMA nodes system.
+
+### Graduation Criteria of Options
+
+In 1.26 we are releasing this feature to Alpha. We propose the following management of TopologyManager policy options graduation:
+
+- `TopologyManagerPolicyAlphaOptions` is not enabled by default. Topology Manager alpha options (only one as of 1.26), are hidden by default
+ and only when `TopologyManagerPolicyAlphaOptions` feature gate is enabled user is able to use alpha options.
+  `TopologyManagerPolicyAlphaOptions` will not exist if there are no options in alpha stage.
+
+- `TopologyManagerPolicyBetaOptions` is not enabled by default. Topology Manager beta options (none as of 1.26), are hidden by default
+ and only when `TopologyManagerPolicyBetaOptions` feature gate is enabled user is able to use beta options.
+`TopologyManagerPolicyBetaOptions` will not exist if there are no options in beta stage (true for 1.26 release).
+
+- `TopologyManagerPolicyOptions` for enabling/disabling the entire feature. As this is an alpha feature, this feature gate would be disabled by default.
+Explicitly enabling `TopologyManagerPolicyOptions` feature gate provides us the ability to supply `TopologyManagerPolicyOptions` or `topology-manager-policy-options` flag in Kubelet.
+
+When an option graduates, its visibility should be moved to be controlled by the corresponding feature-flag.
+The introduction of these feature gates gives us the ability to move the option to beta and later stable without implying that all available options are stable.
+
```

Comment:
I would add a reference to how there is existing precedence for doing the same in the cpumanager here.

Reply:
added